src/imap/mod.rs

```diff
@@ -733,6 +734,53 @@ impl Imap {
         Ok(read_cnt > 0)
     }
 
+    /// Gets the from, to and bcc addresses from all existing outgoing emails.
+    pub async fn get_all_receipients(&mut self, context: &Context) -> Result<Vec<SingleInfo>> {
+        if self.session.is_none() {
+            bail!("IMAP No Connection established");
+        }
+
+        let session = self.session.as_mut().unwrap();
+        let self_addr = context
+            .get_config(Config::ConfiguredAddr)
+            // TODO what if ConfiguredAddr is only username, not username@domain.org?
+            .await
+            .ok_or_else(|| format_err!("Not configured"))?;
+
+        let search_command = format!("FROM \"{}\"", self_addr);
+
+        let uids = session.uid_search(search_command).await?;
+        let uid_vec: Vec<String> = uids.into_iter().map(|s| s.to_string()).collect();
+        let uid_set = uid_vec.join(",");
+        // TODO what if this is, like, 1000 uids? Is there some upper bound for the uid_set when fetching?
+
+        let mut list = session
+            .uid_fetch(uid_set, "(UID BODY.PEEK[HEADER.FIELDS (FROM TO CC BCC)])")
```

Comment:
Postfix drops Bcc: header, afaik because some clients send it in their mails: http://www.postfix.org/postconf.5.html#message_drop_headers

Also I have messages with Bcc: header in my Sent folder. Probably if messages are configured to be saved over IMAP to the inbox folder, they will contain Bcc: too.